.github/workflows/ci.yml

```diff
@@ -1,109 +1,52 @@
 name: CI
 
-on:
-  push:
-    branches:
-      - "*"
-  pull_request:
-    branches:
-      - "*"
-
-env:
-  CARGO_TERM_COLOR: always
-
-defaults:
-  run:
-    shell: bash
+on: [push, pull_request]
 
 jobs:
-  pr:
-    name: pull_request
-    if: "github.event_name == 'pull_request'"
-    strategy:
-      matrix:
-        os:
-          - ubuntu-20.04
-        toolchain:
-          - stable
-          - nightly
-    runs-on: ${{ matrix.os }}
-    needs: lint
+  check:
+    name: Run check
+    runs-on: ubuntu-latest
     steps:
-      - name: checkout the source code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.toolchain }}
+          toolchain: nightly-2021-07-05
           override: true
-      - name: check
-        uses: actions-rs/cargo@v1
+      - run: rustup target add wasm32-unknown-unknown --toolchain nightly-2021-07-05
+      - uses: actions-rs/cargo@v1
         with:
           command: check
-      - name: test
-        if: ${{ matrix.toolchain == 'stable' }}
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-  master:
-    name: push
-    if: "github.event_name == 'push'"
-    strategy:
-      matrix:
-        os:
-          - ubuntu-20.04
-          - windows-2019
-          - macos-10.15
-        toolchain:
-          - stable
-          - beta
-          - nightly
-    runs-on: ${{ matrix.os }}
-    needs: lint
+
+  test:
+    name: Run unit test suite
+    runs-on: ubuntu-latest
     steps:
-      - name: checkout the source code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.toolchain }}
+          toolchain: nightly-2021-07-05
```

Comment:
I'm a bit uneasy about having this as a fixed version. Could we replace this with the equivalent of "use latest", or is that either technically unfeasible or just unwise for stability reasons?

Reply:
there is a bug atm: https://github.community/t/failed-sending-data-to-the-peer-connection-died-tried-5-times-before-giving-up/189130 
But maybe having a fixed nightly is not a bad idea anyway (builds are reproducible)?

Reply:
Yup, reproducibility is a very good argument, especially if nightly is sufficiently unstable for such bugs. I would say we still want to have some procedures around bumping the version once in a while, could you write this down somewhere so we don't forget? I'm trying to avoid a situation in which it turns out we are using a year old version of "nightly" that is actually behind the stable compiler. :P

Reply:
any ideas where/how, is wiki a good place?

Reply:
Yeah, I think somewhere there, but I'm not sure where exactly. For now at top level of "procedures" I guess – we are unlikely to forget about it then.